modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java

```diff
@@ -65,61 +67,122 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 	}
 
-	private Object _invokeConfigurationOverride(Method method, Object[] args)
-		throws IllegalAccessException, InvocationTargetException,
-			   NoSuchMethodException {
+	private Object _getValueNoDefault(Class<?> returnType, String key)
```

Comment:
This method doesn't seem right to me as it always returns a string type value. Also I'd compress _getValueNoDefault and _getValueWithDefault to just a single method with a boolean for whether to use the 2 arg version of the method or not.